### ISSUE TITLE
Github merge refs are eventually consistent

### ISSUE THREAD

**Issue by timperrett** [seek-oss/github-merged-pr-buildkite-plugin#10]
Thanks for this plugin - it generally works well. There is however a problem in that github merge refs are eventually consistent: they do not exist as soon as the HEAD ref is pushed (presumably they have some async job that runs on their side, which would make sense). In this case, when this plugin is active, what happens is that Buildkite is unable to checkout the merge refs, because they presently do not exist. 

Ideally, this script would do an existence check before trying to propagate the merge refs to the build for checkout. Have you experienced this also?

**Comment by zsims** [comment]
We haven't seen this -- our builds must not kick off as fast (at least in most cases).

Do you think it should wait for the ref to exist? Or only kick off a build if it does exist?

**Comment by timperrett** [comment]
FWIW, I would not expect to see this on Github enterprise, but it appears common on github.com. The outlier cases are having to wait about a minute - most everything converges quicker than that, which makes me think simply polling to check for the existence of the merge ref would be the way to go, and only incur a small cost whilst covering an important edge case. WDYT?

**Comment by zsims** [comment]
So apparently this changed very recently: https://discourse.drone.io/t/github-claims-that-merge-refs-are-undocumented-feature/1100

TL;DR: the merge check is no longer automatic, and is done on visiting the PR page. 

Polling would be difficult to do, as a minute may not be long enough (if no one visits the page). I'll look at changing this plugin to do a merge explicitly (which is what GitHub do anyway)

**Comment by timn** [comment]
Hi. Thanks for the plugin! We are interested in using it. Has there been any progress towards explicit merges?

**Comment by zsims** [comment]
Hi, thanks for checking it out. Hasn't been any progress towards an explicit merge -- I'll give it a crack to see what's possible

**Comment by zsims** [comment]
@timn I've pushed out a release `v0.0.6` that does the explicit merge. I've done a bit of testing -- seems to work well

Let me know how it goes for you :-)

**Comment by timn** [comment]
Thanks @zsims. Will give it a try tomorrow!

**Comment by timn** [comment]
@zsims, gave this a try, works like a charm. Thanks!

Have you tried this for merges with conflicts (I haven't, yet)? Will the build fail in that case?

**Comment by zsims** [comment]
Yeah, it fails the PR if there's conflicts -- but only in `checkout` mode. `trigger` mode kicks off another build, and Buildkite doesn't report the status back to GitHub (about the build failing).

TL;DR just use `checkout` mode 😄 

![screen shot 2018-10-01 at 9 28 56 am](https://user-images.githubusercontent.com/1764368/46264148-78412b00-c55c-11e8-8f84-ea5f25a2104a.png)
